### PR TITLE
Wismy/x10(s) upgrade kit

### PR DESCRIPTION
**Pull request by WismyYao**
add a popup for auth-failure

### CODE REVIEW

File: radio/src/telemetry/frsky_pxx2.cpp [hunk 1]
```diff
@@ -227,6 +227,11 @@ void processAuthenticationFrame(uint8_t module, uint8_t * frame)
   uint8_t cryptoType = frame[3];
   uint8_t messageDigest[16] = {0};
 
+  if (0x04 == frame[0] && 0xA5 == frame[4]) {
```
Comment (raphaelcoeffic) [inline radio/src/telemetry/frsky_pxx2.cpp]: Could you please add “defines” for your constants?

Comment (WismyYao) [reply radio/src/telemetry/frsky_pxx2.cpp]: Which file should I put the 'defines' in ? How about the strings ?

Comment (raphaelcoeffic) [reply radio/src/telemetry/frsky_pxx2.cpp]: `pulses/pxx2.h` would be a good place, next to the other frame defs. Thx!

Comment (bsongis) [reply radio/src/telemetry/frsky_pxx2.cpp]: I will take it

File: radio/src/telemetry/frsky_pxx2.cpp [hunk 2]
```diff
@@ -227,6 +227,11 @@ void processAuthenticationFrame(uint8_t module, uint8_t * frame)
...
+    POPUP_INFORMATION("Auth-failure");
```
Comment (bsongis) [inline radio/src/telemetry/frsky_pxx2.cpp]: Missing translations

File: radio/src/translations/de.h.txt [hunk 3]
```diff
@@ -1157,6 +1157,11 @@
 #define TR_MENU_INVERT                 "Invertieren<!>"
 #define TR_JITTER_FILTER               "ADC Filter"
 #define TR_RTC_CHECK                   TR("Check RTC", "Check RTC voltage")
+
+#if defined(ACCESS_LIB)
```
Comment (bsongis) [inline radio/src/translations/de.h.txt]: We don't add so many ifdef here

File: radio/src/translations.h [hunk 4]
```diff
@@ -910,6 +910,10 @@ extern const char STR_ABOUT_PARENTS_2[];
 extern const char STR_ABOUT_PARENTS_3[];
 extern const char STR_ABOUT_PARENTS_4[];
 
+#if defined(ACCESS_LIB)
```
Comment (bsongis) [inline radio/src/translations.h]: The compiler can optimize this if not needed

File: radio/src/translations.cpp [hunk 5]
```diff
@@ -770,3 +770,7 @@ const char STR_ABOUT_PARENTS_1[] = TR_ABOUT_PARENTS_1;
 const char STR_ABOUT_PARENTS_2[] = TR_ABOUT_PARENTS_2;
 const char STR_ABOUT_PARENTS_3[] = TR_ABOUT_PARENTS_3;
 const char STR_ABOUT_PARENTS_4[] = TR_ABOUT_PARENTS_4;
+
+#if defined(ACCESS_LIB)
```
Comment (bsongis) [inline radio/src/translations.cpp]: Here as well, no if defined

File: radio/src/telemetry/frsky_pxx2.cpp (hunk 1 above)
Comment (bsongis) [reply radio/src/telemetry/frsky_pxx2.cpp]: Now I read it more carefully, the 0x04 is not a flag, it's just a sanity check, I have changed it to `frame[0] >= 4`

**Pull request merged**